sequencer/interfaces.go

```diff
@@ -29,25 +29,42 @@ type etherman interface {
 
 // stateInterface gathers the methods required to interact with the state.
 type stateInterface interface {
+	OpenBatch(ctx context.Context, processingContext state.ProcessingContext, dbTx pgx.Tx) error
 	GetLastVirtualBatchNum(ctx context.Context, dbTx pgx.Tx) (uint64, error)
 	GetTimeForLatestBatchVirtualization(ctx context.Context, dbTx pgx.Tx) (time.Time, error)
 	GetTxsOlderThanNL1Blocks(ctx context.Context, nL1Blocks uint64, dbTx pgx.Tx) ([]common.Hash, error)
 	GetBatchByNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) (*state.Batch, error)
 	GetTransactionsByBatchNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) (txs []types.Transaction, err error)
 	IsBatchClosed(ctx context.Context, batchNum uint64, dbTx pgx.Tx) (bool, error)
+	Begin(ctx context.Context) (pgx.Tx, error)
+	ProcessSingleTx(request state.ProcessSingleTxRequest) state.ProcessBatchResponse
+	BeginStateTransaction(ctx context.Context) (pgx.Tx, error)
 }
 
 type txManager interface {
 	SequenceBatches(ctx context.Context, sequences []ethmanTypes.Sequence) error
 }
 
 type workerInterface interface {
-	GetBestFittingTx(resources RemainingResources) *TxTracker
+	GetBestFittingTx(resources BatchResources) *TxTracker
+	UpdateAfterSingleSuccessfulTxExecution(from common.Address, touchedAddresses map[common.Address]*state.TouchedAddress)
+	UpdateTx(txHash common.Hash, from common.Address, ZKCounters state.ZKCounters)
 }
 
+// The dbManager will need to handle the errors inside the functions which don't return error as they will be used async in the other abstractions.
+// Also if dbTx is missing this needs also to be handled ion the dbManager
```

Comment:
```suggestion
// Also if dbTx is missing this needs also to be handled in the dbManager
```

What do you mean? I think all method should have dbTx in their signature